packages/ocvImpact/R/utils_targeting.R

```diff
@@ -104,102 +113,169 @@ load_targets_by_country <- function(datapath, modelpath, country){
+  
+  wide_coverage <- coverage %>% tidyr::pivot_wider(names_from = vaccine, values_from = coverage)
+  ocv2_coverage <- wide_coverage$OCV2[!is.na(wide_coverage$OCV2)]
+  wide_coverage <- subset(wide_coverage, (!is.na(wide_coverage$OCV1)))
+  wide_coverage$OCV2 <- ocv2_coverage
+  coverage <- wide_coverage
+  rm(wide_coverage) ##wide_coverage was just a template, remove to save memory
   
   coverage_as_all_0_for_campaign <- (sum(coverage$coverage) == 0)
+  
   if (!coverage_as_all_0_for_campaign){
+    
     coverage <- import_coverage_scenario(modelpath, country, scenario, filter0 = TRUE, redownload = FALSE)
+    
+    ##added procedure to add ocv2 rows for the coverage for the ocv1 scenario to ensure implementation is consistent across scenarios
+    
+    if (runname == "202310gavi-4" & ndoses == "one"){
+      
+      coverage_copy <- coverage
+      coverage_copy$vaccine <- 'OCV2'
+      coverage_copy$coverage <- 0 ##ocv2 rows get 0 coverage for the ocv1 scenario
+      coverage <- rbind(coverage, coverage_copy)
+      rm(coverage_copy) ##coverage_copy was just a template, remove to save memory
+      
+    }
+    
+    coverage <- adjusted_montagu_coverage(coverage_sheet = coverage, country = country)
+    
+    ##procedure to make coverage dataframe 'wider' (get new columns for ocv1 and ocv2 coverage)
+    
+    wide_coverage <- coverage %>% tidyr::pivot_wider(names_from = vaccine, values_from = coverage)
+    ocv2_coverage <- wide_coverage$OCV2[!is.na(wide_coverage$OCV2)]
+    wide_coverage <- subset(wide_coverage, (!is.na(wide_coverage$OCV1)))
+    wide_coverage$OCV2 <- ocv2_coverage
+    coverage <- wide_coverage
+    rm(wide_coverage) ##wide_coverage was just a template, remove to save memory
+    
   }
-
+  
   if (is.null(coverage)){
-
+    
     message("No vaccine was assigned for the no-vaccination scenario. Returning NULL value.")
     ftargets_flat <- NULL
-  
+    
   } else{
-
+    
     ## Perform checks on the coverage scenario
     if (!all(coverage$gender == "both") | 
         !all(coverage$age_range_verbatim == "default age and gender" | coverage$age_range_verbatim == ">1y" | coverage$age_range_verbatim == "default age groups" | coverage$age_range_verbatim == "<NA>" | coverage$age_range_verbatim == "1-100")|
         !all(coverage$activity_type == "campaign")
-        ){
+    ){
       stop(paste("Vaccine assignment is not supported for this coverage scenario. Check the gender, age_range_verbatim, and activity_type columns in the", scenario, "coverage sheet."))
     }
-
+    
     ftargets <- vector(mode = "list", length = length(coverage$year))
-
+    
     for (i in 1:length(coverage$year)){
-
+      
       cov_year <- coverage[i,]
       goal_target_pop <- cov_year$target
-      goal_fvp <- cov_year$fvp
+      
+      ##calculate fvps with one dose and two doses of the vaccine
+      goal_ocv1_fvp <- cov_year$OCV1 * goal_target_pop
```

Comment:
Can we go over these quickly together? Just want to make sure I understand...

Reply:
Sure! Should I use our zoom link?